#! /bin/sh -e
## 01_fix-mono-calls.dpatch by Michael Schiansky <ms@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix all cli/mono calls (See: #265521)
## DP: thanks to: Dave Beckett <dave.beckett@bristol.ac.uk>

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
diff -urNad /home/ms/uprojects/pkg/mono/build/1.0.2/mcs/mcs-1.0.2/gmcs/Makefile mcs-1.0.2/gmcs/Makefile
--- /home/ms/uprojects/pkg/mono/build/1.0.2/mcs/mcs-1.0.2/gmcs/Makefile	2004-06-16 19:13:41.000000000 +0200
+++ mcs-1.0.2/gmcs/Makefile	2004-09-24 17:47:42.000000000 +0200
@@ -69,7 +69,7 @@
 dist-local: dist-default
 
 gmcs.exe: $(all_sources)
-	MONO_PATH="../class/lib/net_2_0_bootstrap$(PLATFORM_PATH_SEPARATOR)$$MONO_PATH" mono ../mcs/mcs.exe $(DEBUG_FLAGS) -2 /target:exe /debug /out:$@ $^
+	MONO_PATH="../class/lib/net_2_0_bootstrap$(PLATFORM_PATH_SEPARATOR)$$MONO_PATH" $(RUNTIME) ../mcs/mcs.exe $(DEBUG_FLAGS) -2 /target:exe /debug /out:$@ $^
 
 gmcs2.exe: $(all_sources)
 	MONO_PATH="../class/lib/net_2_0$(PLATFORM_PATH_SEPARATOR)$$MONO_PATH" time mcs -2 /target:exe /debug /out:$@ $^ $(DEBUG_FLAGS) 
@@ -110,12 +110,12 @@
 	@ $(FRIENDLY_TIME) $(RUNTIME) ./mcs2.exe $(USE_MCS_FLAGS) /target:exe /out:mcs3.exe $(all_sources) > /dev/null || (echo FAILED; exit 1)
 	@ echo -n "corlib:  "
 	@ rm -f ../class/lib/mscorlib.dll
-	@ (cd ../class/corlib ; make BOOTSTRAP_MCS="$(FRIENDLY_TIME) mono ../../mcs/mcs.exe" > /dev/null ) || (echo FAILED; exit 1)
+	@ (cd ../class/corlib ; make BOOTSTRAP_MCS="$(FRIENDLY_TIME) $(RUNTIME) ../../mcs/mcs.exe" > /dev/null ) || (echo FAILED; exit 1)
 
 do-corlib:
 	@ echo -n "corlib:  "
 	@ rm -f ../class/lib/mscorlib.dll
-	@ (cd ../class/corlib ; make BOOTSTRAP_MCS="$(FRIENDLY_TIME) mono ../../mcs/mcs.exe" > /dev/null ) || (echo FAILED; exit 1)
+	@ (cd ../class/corlib ; make BOOTSTRAP_MCS="$(FRIENDLY_TIME) $(RUNTIME) ../../mcs/mcs.exe" > /dev/null ) || (echo FAILED; exit 1)
 
 PROFILER=default
 
diff -urNad /home/ms/uprojects/pkg/mono/build/1.0.2/mcs/mcs-1.0.2/mcs/Makefile mcs-1.0.2/mcs/Makefile
--- /home/ms/uprojects/pkg/mono/build/1.0.2/mcs/mcs-1.0.2/mcs/Makefile	2004-05-24 12:37:00.000000000 +0200
+++ mcs-1.0.2/mcs/Makefile	2004-09-24 17:47:36.000000000 +0200
@@ -62,7 +62,7 @@
 do-corlib:
 	@ echo -n "corlib:  "
 	@ rm -f ../class/lib/mscorlib.dll
-	@ cd ../class/corlib ; $(MAKE) BOOTSTRAP_MCS="$(FRIENDLY_TIME) mono ../../mcs/mcs.exe" > /dev/null || (echo FAILED; exit 1)
+	@ cd ../class/corlib ; $(MAKE) BOOTSTRAP_MCS="$(FRIENDLY_TIME) $(RUNTIME) ../../mcs/mcs.exe" > /dev/null || (echo FAILED; exit 1)
 
 PROFILER=default
 
